src/signal.rs

```diff
@@ -474,4 +484,46 @@ mod tests {
         // forever as the loop keeps running. Since we don't join, the thread will become detached
         // as the handle is dropped, and will be killed when the process/main thread exits.
     }
+
+    #[test]
+    fn test_block_unblock_signal() {
+        let signal = SIGRTMIN() + 0;
+
+        // Check if it is blocked.
+        unsafe {
+            let mut sigset: sigset_t = mem::zeroed();
+            pthread_sigmask(SIG_BLOCK, null(), &mut sigset as *mut sigset_t);
+            assert_eq!(sigismember(&sigset, signal), 0);
+        }
+
+        block_signal(signal).unwrap();
+        assert!(get_blocked_signals().unwrap().contains(&(signal)));
+
+        unblock_signal(signal).unwrap();
+        assert!(!get_blocked_signals().unwrap().contains(&(signal)));
+    }
+
+    #[test]
+    fn test_clear_pending() {
+        let signal = SIGRTMIN() + 1;
+
+        block_signal(signal).unwrap();
+
+        // Block the signal, which means it won't be delivered until it is
+        // unblocked. Pending between the time when the signal which is set as blocked
+        // is generated and when is delivered.
+        let killable = thread::spawn(move || {
+            loop {
+                // Wait for the signal being killed.
+                thread::sleep(Duration::from_millis(100));
+                if is_pending(signal) {
+                    clear_signal(signal).unwrap();
+                }
+                assert!(!is_pending(signal));
```

Comment:
I think this assert should be moved after `clear_signal` inside the `if` block, something like this:

```rust
if is_pending(signal) {
    clear_signal(signal).unwrap();
    assert!(!is_pending(signal));
    break;
}
```
I've added the break such that the thread will exit after clearing the signal.

Reply:
Addressed.